homeassistant/components/google_assistant/helpers.py

```diff
@@ -450,25 +474,10 @@ async def sync_serialize(self, agent_user_id):
         room = entity_config.get(CONF_ROOM_HINT)
         if room:
             device["roomHint"] = room
-            return device
-
-        dev_reg, ent_reg, area_reg = await gather(
-            self.hass.helpers.device_registry.async_get_registry(),
-            self.hass.helpers.entity_registry.async_get_registry(),
-            self.hass.helpers.area_registry.async_get_registry(),
-        )
-
-        entity_entry = ent_reg.async_get(state.entity_id)
-        if not (entity_entry and entity_entry.device_id):
-            return device
-
-        device_entry = dev_reg.devices.get(entity_entry.device_id)
-        if not (device_entry and device_entry.area_id):
-            return device
 
-        area_entry = area_reg.areas.get(device_entry.area_id)
-        if area_entry and area_entry.name:
-            device["roomHint"] = area_entry.name
+        area = await _get_area(self.hass, state.entity_id)
+        if area and area.name:
```

Comment:
This seems incorrect. If a configuration is overriding the room hint, this will now actually override the explicitly configured room hint by the user.

Reply:
Indeed. Resolved